### ISSUE TITLE
clang compiler error on Travis: "unable to copy EFLAGS physical register"

### ISSUE THREAD

**Issue by derekbruening** [DynamoRIO/dynamorio#3989]
We have hit this a few times now:

https://travis-ci.com/DynamoRIO/dynamorio/builds/139582682

https://travis-ci.com/DynamoRIO/dynamorio/jobs/266673736

https://travis-ci.com/DynamoRIO/dynamorio/jobs/263543026

As pointed out in #3975 :

It looks like it's known and fixed in recent clang:
https://bugs.llvm.org/show_bug.cgi?id=38865

I have not looked for a filed Travis bug or how easy it is to upgrade clang in our script.

**Comment by derekbruening** [comment]
So our Travis Xenial has clang 3.7.  The fix we need is r341972 or bettter, from 2018-09-24.  That requires clang 7.0.  This page https://docs.travis-ci.com/user/languages/c/ guarantees that clang 5.0 will install easily as a binary package, but doesn't list anything higher.  We can try clang-7.

**Comment by derekbruening** [comment]
Looking at an instance of this problem at https://travis-ci.com/DynamoRIO/dynamorio/jobs/266715592 it says we already have clang-7 (maybe b/c we're using `llvm-toolchain-xenial-7` but that's just a source):

```
$ clang --version
clang version 7.0.0 (tags/RELEASE_700/final)

debug-internal-64: **** 7 build errors ****
	fatal error: error in backend: Unable to copy EFLAGS physical register!
	clang-7: error: clang frontend command failed with exit code 70 (use -v to see invocation)
	core/CMakeFiles/unit_tests.dir/build.make:1138: recipe for target 'core/CMakeFiles/unit_tests.dir/arch/x86_code_test.c.o' failed
	CMakeFiles/Makefile2:1446: recipe for target 'core/CMakeFiles/unit_tests.dir/all' failed
	make[2]: *** [core/CMakeFiles/unit_tests.dir/arch/x86_code_test.c.o] Error 70
	make[1]: *** [core/CMakeFiles/unit_tests.dir/all] Error 2
	make: *** [all] Error 2
```

Plus, that's a 64-bit build, while the clang bug tracker entry cited above claims to be 32-bit only.
Looking further there is another entry:

https://bugs.llvm.org/show_bug.cgi?id=39073
Says bug is there in 7, fixed in later version.

**Comment by derekbruening** [comment]
Well I can successfully get Travis to use clang 8.0 but that hits a similar problem:
https://travis-ci.com/DynamoRIO/dynamorio/jobs/267069332
```
-- The C compiler identification is Clang 8.0.1
-- Check for working C compiler: /usr/bin/clang-8 -- works
-- The CXX compiler identification is Clang 8.0.1
-- Check for working CXX compiler: /usr/bin/clang++-8
<...>
[ 37%] Building C object core/CMakeFiles/unit_tests.dir/arch/x86_code_test.c.o
<...>
fatal error: error in backend: Cannot emit physreg copy instruction
clang: error: clang frontend command failed with exit code 70 (use -v to see invocation)
clang version 8.0.1-svn369350-1~exp1~20190820122438.78 (branches/release_80)
Target: i386-pc-linux-gnu
Thread model: posix
InstalledDir: /usr/bin
clang: note: diagnostic msg: PLEASE submit a bug report to https://bugs.llvm.org/ and include the crash backtrace, preprocessed source, and associated run script.
[ 37%] Built target client.drwrap-test.appdll
[ 37%] Linking CXX executable bin/client.drwrap-test-callconv
Scanning dependencies of target client.drutil-test
[ 37%] Building C object suite/tests/CMakeFiles/client.drutil-test.dir/client-interface/drutil-test.c.o
clang: note: diagnostic msg: 
********************

PLEASE ATTACH THE FOLLOWING FILES TO THE BUG REPORT:
Preprocessed source(s) and associated run script(s) are located at:
clang: note: diagnostic msg: /tmp/x86_code_test-1ef0d8.c
clang: note: diagnostic msg: /tmp/x86_code_test-1ef0d8.sh
clang: note: diagnostic msg: 

********************
core/CMakeFiles/unit_tests.dir/build.make:1125: recipe for target 'core/CMakeFiles/unit_tests.dir/arch/x86_code_test.c.o' failed
CMakeFiles/Makefile2:1377: recipe for target 'core/CMakeFiles/unit_tests.dir/all' failed
make[2]: *** [core/CMakeFiles/unit_tests.dir/arch/x86_code_test.c.o] Error 70

<...>
debug-internal-32: **** 7 build errors ****
	fatal error: error in backend: Cannot emit physreg copy instruction
	clang: error: clang frontend command failed with exit code 70 (use -v to see invocation)
	core/CMakeFiles/unit_tests.dir/build.make:1125: recipe for target 'core/CMakeFiles/unit_tests.dir/arch/x86_code_test.c.o' failed
```

Wow, is our x86_code_test.c really that strange to trigger multiple compiler bugs??

**Comment by derekbruening** [comment]
OK that one was reported by @hgreving2304: https://bugs.llvm.org/show_bug.cgi?id=41678
Fixed in r359837 but we need 9.0 for that.
